### PR TITLE
Add OAuth2 with Proxy AuthN prd test environment

### PR DESCRIPTION
**Pull request by marcinsbd**
## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthn.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import io.trino.tests.product.launcher.env.common.TestsEnvironment;
+import io.trino.tests.product.launcher.testcontainers.PortBinder;
+
+import static io.trino.tests.product.launcher.env.EnvironmentContainers.COORDINATOR;
+import static io.trino.tests.product.launcher.env.common.Standard.CONTAINER_TRINO_CONFIG_PROPERTIES;
+import static org.testcontainers.utility.MountableFile.forHostPath;
+
+@TestsEnvironment
+public class EnvSinglenodeOauth2HttpProxyAuthn
+        extends EnvSinglenodeOauth2HttpProxy
```
Comment (Praveen2112) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthn.java:31]: Can we avoid extending existing `Environments` ? In case of reusing, we could have some sort of an `EnvironmentExtender` and capture related code in that extender

File: testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/singlenode-oauth2-https-proxy-authn/httpd.conf [hunk 2]
```diff
@@ -0,0 +1,62 @@
+#
```
Comment (Praveen2112) [inline testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/singlenode-oauth2-https-proxy-authn/httpd.conf:1]: Can we have a dedicated `EnvironmentExtender` for Proxy ?

**Comment by Praveen2112** [comment]
Since the latest arilift wasn't updated - The tests should fail right ?

**Comment by marcinsbd** [comment]
> Since the latest arilift wasn't updated - The tests should fail right ?
Proxy Authn is added within Airlift v259
Trino was updated today to Airlift v259

**Comment by Praveen2112** [comment]
This PR was using ?

**Comment by marcinsbd** [comment]
> This PR was using ?

tldr
PR used version with airlift 259

from the test log: (https://github.com/trinodb/trino/actions/runs/10283448875/job/28457710590?pr=22934)
` HEAD is now at ce70f3f58d Merge 4992116feab94f5eac08f08a091eded96697a4f2 into 07c4f7cd5ebb0320b8047c6b42f86630d02576eb`
07c4f7cd5ebb0320b8047c6b42f86630d02576eb.  is that one
https://github.com/trinodb/trino/commit/07c4f7cd5ebb0320b8047c6b42f86630d02576eb
which was added after airlift upgrade to v259

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxy.java [hunk 3]
```diff
@@ -82,9 +82,5 @@ public void extendEnvironment(Environment.Builder builder)
         builder.containerDependsOn(COORDINATOR, hydraClientConfig.getLogicalName());
 
         builder.containerDependsOn(COORDINATOR, PROXY);
-        builder.configureContainer(PROXY, dockerContainer -> {
-            dockerContainer
-                    .withCopyFileToContainer(forHostPath(configDir.getPath("httpd.conf")), "/usr/local/apache2/conf/httpd.conf");
-        });
```
Comment (Praveen2112) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxy.java]: This change has to be squahsed to the previous one right ?

Comment (marcinsbd) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxy.java]: yes,  done

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxy.java [hunk 4]
```diff
@@ -75,9 +81,10 @@ public void extendEnvironment(Environment.Builder builder)
 
         builder.containerDependsOn(COORDINATOR, hydraClientConfig.getLogicalName());
 
-        DockerContainer proxy = new DockerContainer("httpd:2.4.51", "proxy");
-        proxy.withCopyFileToContainer(forHostPath(configDir.getPath("httpd.conf")), "/usr/local/apache2/conf/httpd.conf");
-        builder.addContainer(proxy);
-        builder.containerDependsOn(COORDINATOR, "proxy");
+        builder.containerDependsOn(COORDINATOR, PROXY);
```
[the remark this answers is not in the document]
Comment (marcinsbd) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxy.java:84]: So in this way, I should add proxy container as the first argument to the `super` constructor?

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpsProxy.java [hunk 5]
```diff
@@ -78,15 +84,13 @@ public void extendEnvironment(Environment.Builder builder)
...
+        builder.configureContainer(PROXY, proxyContainer -> proxyContainer
                 .withCopyFileToContainer(
                         forHostPath(configDir.getPath("httpd.conf")),
                         "/usr/local/apache2/conf/httpd.conf")
                 .withCopyFileToContainer(
                         forHostPath(configDir.getPath("cert")),
-                        "/usr/local/apache2/conf/cert");
-        builder.addContainer(proxy);
-        builder.containerDependsOn(COORDINATOR, "proxy");
+                        "/usr/local/apache2/conf/cert"));
```
Comment (ssheikin) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpsProxy.java]: extract path as a constant to the env extender

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/HttpsProxy.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+                .getDockerFilesHostDirectory("common/https-proxy");
+        this.httpProxy = requireNonNull(httpProxy, "httpProxy is null");
+    }
+
+    @Override
+    public void extendEnvironment(Environment.Builder builder)
+    {
+        builder.configureContainer(PROXY, proxyContainer -> proxyContainer
+                .withCopyFileToContainer(
+                        forHostPath(configDir.getPath("httpd.conf")),
```
Comment (ssheikin) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/HttpsProxy.java:48]: could you please move the source file to the same location `/usr/local/apache2/conf` in the git repository, as it will be mapped in the container?

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+            PortBinder binder,
+            Standard standard,
+            HydraIdentityProvider hydraIdentityProvider,
+            HttpProxy httpProxy)
+    {
+        super(standard, hydraIdentityProvider, httpProxy);
+
+        this.binder = requireNonNull(binder, "binder is null");
+        this.hydraIdentityProvider = requireNonNull(hydraIdentityProvider, "hydraIdentityProvider is null");
+        requireNonNull(dockerFiles, "dockerFiles is null");
```
Comment (ssheikin) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java:55]: inline to the next line.

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+            binder.exposePort(dockerContainer, 7778);
+        });
+
+        DockerContainer hydraClientConfig = hydraIdentityProvider.createClient(
+                builder,
+                "trinodb_client_id",
+                "trinodb_client_secret",
+                "client_secret_basic",
+                "trinodb_client_id/",
```
Comment (ssheikin) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java:79]: is `/` needed?

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+                "https://presto-master:7778/oauth2/callback,https://localhost:7778/oauth2/callback");
+
+        builder.containerDependsOn(COORDINATOR, hydraClientConfig.getLogicalName());
+
+        builder.containerDependsOn(COORDINATOR, PROXY);
+        builder.configureContainer(PROXY, dockerContainer -> {
+            dockerContainer
+                    .withCopyFileToContainer(forHostPath(configDir.getPath("httpd.conf")), "/usr/local/apache2/conf/httpd.conf");
```
Comment (ssheikin) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java:87]: please extract `/usr/local/apache2/conf/` to constant

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+{
+    private final PortBinder binder;
+    private final HydraIdentityProvider hydraIdentityProvider;
+    private final ResourceProvider configDir;
+
+    @Inject
+    public EnvSinglenodeOauth2HttpProxyAuthentication(
+            DockerFiles dockerFiles,
+            PortBinder binder,
+            Standard standard,
```
Comment (ssheikin) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java:47]: can we make multinode?

File: testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/singlenode-oauth2-https-proxy-authn/.htpasswd [hunk 11]
```diff
@@ -0,0 +1 @@
+proxy_user:$apr1$M4Whksmo$csx/0NiGcPVE06m3AyQfz.
```
Comment (ssheikin) [inline testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/singlenode-oauth2-https-proxy-authn/.htpasswd:1]: please move files not related to trino to `proxy` directory:
`singlenode-oauth2-https-proxy-authn/proxy/.htpasswd`

File: testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/singlenode-oauth2-https-proxy-authn/config.properties [hunk 12]
```diff
@@ -0,0 +1,30 @@
+node.id=will-be-overwritten
+node.environment=test
+
```
Comment (ssheikin) [inline testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/singlenode-oauth2-https-proxy-authn/config.properties:3]: please move files related to trino to `trino` directory:
`singlenode-oauth2-https-proxy-authn/trino/config.properties`

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/HttpsProxy.java (hunk 6 above)
Comment (marcinsbd) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/HttpsProxy.java:49]: imho,  moving to just subdir proxy will be just enough as there are no others files that need to be transferred to other dirs within proxy container, won't be just too overeagerly?   what do others think?

Comment (ssheikin) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/HttpsProxy.java:49]: later we can consider to bind folder from git repository to folder in container, 
and the structure will follow.

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpsProxy.java (hunk 5 above)
Comment (ssheikin) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpsProxy.java]: Either https://github.com/trinodb/trino/pull/22934/commits/0faaf8a784a459bcd7f8612130532d265a52faa4#r1713582652 is not resolver, or it's not pushed, or it fixed in the wrong commit.

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2.java [hunk 13]
```diff
@@ -70,7 +70,7 @@ public void extendEnvironment(Environment.Builder builder)
                 "trinodb_client_id",
                 "trinodb_client_secret",
                 "client_secret_basic",
-                "trinodb_client_id/",
```
Comment (ssheikin) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2.java:73]: If possible it's always nice to keep refactoring commits as first ones within PR.

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpsProxy.java (hunk 5 above)
Comment (marcinsbd) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpsProxy.java]: This one is Env\*Http**s**Proxy, that one was Env\*HttpProxy, nvm, fixed

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/HttpsProxy.java (hunk 6 above)
Comment (marcinsbd) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/HttpsProxy.java:49]: conf directory contains also other files that we would need to incorporate to git, for now I don't see any advantage, let's keep it simple

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java (hunk 10 above)
Comment (marcinsbd) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java]: I rollbacked to singlenode, as imho the coordinator is responsible for communication with clients
I think adding workers does not add any value for the tests, does it?

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2.java (hunk 13 above)
Comment (marcinsbd) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2.java:73]: yes, I agree, however I thought the commit is not directly connected with the change ( as it can be added before or after without the influence on the actual change in PR) so I can leave as it is
on the other hand I can reshuffle

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java (hunk 10 above)
Comment (marcinsbd) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java:48]: Let's rewrite it as multinode in follow-up ticket

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxy.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (Praveen2112) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxy.java:84]: Maybe but I guess we could have it as separate PR

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/EnvironmentModule.java
Comment (Praveen2112) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/EnvironmentModule.java:1]: Can we squash it with a previous commit ?

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+    public void extendEnvironment(Environment.Builder builder)
+    {
+        builder.configureContainer(COORDINATOR, dockerContainer -> {
+            dockerContainer
+                    .withCopyFileToContainer(
+                            forHostPath(configDir.getPath("trino/config.properties")),
+                            CONTAINER_TRINO_CONFIG_PROPERTIES)
+                    .withCopyFileToContainer(
+                            forHostPath(configDir.getPath("trino/log.properties")),
+                            CONTAINER_TRINO_ETC + "/log.properties");
```
Comment (Praveen2112) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java:70]: Since we have moved everything to `trino` folder I guess we could copy the content of that folder here.

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+        builder.containerDependsOn(COORDINATOR, hydraClientConfig.getLogicalName());
+
+        builder.containerDependsOn(COORDINATOR, PROXY);
+        builder.configureContainer(PROXY, dockerContainer -> {
+            dockerContainer.withCopyFileToContainer(
+                    forHostPath(configDir.getPath("proxy/httpd.conf")),
+                    HTTP_PROXY_CONF_DIR + "httpd.conf");
+            dockerContainer.withCopyFileToContainer(
+                    forHostPath(configDir.getPath("proxy/.htpasswd")),
+                    HTTP_PROXY_CONF_DIR + ".htpasswd");
```
Comment (Praveen2112) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java:92]: Same here as well

**Review by Praveen2112 — COMMENTED** [review]
LGTM

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java (hunk 14 above)
Comment (marcinsbd) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java:70]: I think there is no explicit support for copying whole directory

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/EnvironmentModule.java
Comment (marcinsbd) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/EnvironmentModule.java:1]: Which ones you meant?  Commits 'Extracting httpProxy' and 'Extracting httpsProxy' to squash together?

Comment (Praveen2112) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/EnvironmentModule.java:1]: Yes

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java (hunk 14 above)
Comment (Praveen2112) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java:70]: I think we can copy them via 
```
                    .withCopyFileToContainer(
                            forHostPath(configDir.getPath("trino")),
                            CONTAINER_TRINO_ETC)
```

Comment (marcinsbd) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java]: done

**Comment by marcinsbd** [comment]
@Praveen2112 and @ssheikin thank you for the review,
pls let me know if you think there is sth to improve

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2.java
Comment (Praveen2112) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2.java:1]: How about ?

Remove redundant trailing slash

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/EnvironmentModule.java
Comment (Praveen2112) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/EnvironmentModule.java:1]: Can we reword as 
 Extract HttpProxy as an EnvironmentExtender

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/HttpProxy.java [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+import static org.testcontainers.utility.MountableFile.forHostPath;
+
+public class HttpProxy
+        implements EnvironmentExtender
+{
+    public static final String HTTP_PROXY_CONF_DIR = "/usr/local/apache2/conf";
+    public static final String PROXY = "proxy";
+    private static final String HTTPD_VERSION = "2.4.51";
+
+    private final DockerFiles.ResourceProvider configDir;
```
Comment (Praveen2112) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/HttpProxy.java:31]: nit: static import

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/EnvironmentModule.java
Comment (Praveen2112) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/EnvironmentModule.java:1]: Same for HttpsProxy as well

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java
Comment (Praveen2112) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java:1]: Add environment for OAuth2 with authenticated proxy

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java [hunk 17]
```diff
@@ -0,0 +1,86 @@
...
+                .getDockerFilesHostDirectory("conf/environment/singlenode-oauth2-http-proxy-authn/");
+    }
+
+    @Override
+    public void extendEnvironment(Environment.Builder builder)
+    {
+        builder.configureContainer(COORDINATOR, dockerContainer -> {
+            dockerContainer
+                    .withCopyFileToContainer(
+                            forHostPath(configDir.getPath("trino")), CONTAINER_TRINO_ETC);
```
Comment (Praveen2112) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java:65]: nit: Can we inline them ?

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java [hunk 18]
```diff
@@ -0,0 +1,86 @@
...
+
+import static io.trino.tests.product.launcher.env.EnvironmentContainers.COORDINATOR;
+import static io.trino.tests.product.launcher.env.common.HttpProxy.HTTP_PROXY_CONF_DIR;
+import static io.trino.tests.product.launcher.env.common.HttpProxy.PROXY;
+import static io.trino.tests.product.launcher.env.common.Standard.CONTAINER_TRINO_ETC;
+import static java.util.Objects.requireNonNull;
+import static org.testcontainers.utility.MountableFile.forHostPath;
+
+@TestsEnvironment
+public class EnvSinglenodeOauth2HttpProxyAuthentication
```
Comment (Praveen2112) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java:36]: nit: EnvSinglenodeOauth2AuthenticatedHttpProxy

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpsProxyAuthentication.java [hunk 19]
```diff
@@ -0,0 +1,91 @@
...
+
+import static io.trino.tests.product.launcher.env.EnvironmentContainers.COORDINATOR;
+import static io.trino.tests.product.launcher.env.common.HttpProxy.HTTP_PROXY_CONF_DIR;
+import static io.trino.tests.product.launcher.env.common.HttpProxy.PROXY;
+import static io.trino.tests.product.launcher.env.common.Standard.CONTAINER_TRINO_ETC;
+import static java.util.Objects.requireNonNull;
+import static org.testcontainers.utility.MountableFile.forHostPath;
+
+@TestsEnvironment
+public class EnvSinglenodeOauth2HttpsProxyAuthentication
```
Comment (Praveen2112) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpsProxyAuthentication.java:36]: nit: EnvSinglenodeOauth2AuthenticatedHttpsProxy

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java [hunk 20]
```diff
@@ -0,0 +1,86 @@
...
+            PortBinder binder,
+            Standard standard,
+            HydraIdentityProvider hydraIdentityProvider,
+            HttpProxy httpProxy)
+    {
+        super(standard, hydraIdentityProvider, httpProxy);
+
+        this.binder = requireNonNull(binder, "binder is null");
+        this.hydraIdentityProvider = requireNonNull(hydraIdentityProvider, "hydraIdentityProvider is null");
+        this.configDir = requireNonNull(dockerFiles, "dockerFiles is null")
```
Comment (Praveen2112) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java:55]: nit: rnn is redundant

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpsProxyAuthentication.java [hunk 21]
```diff
@@ -0,0 +1,91 @@
...
+            PortBinder binder,
+            Standard standard,
+            HydraIdentityProvider hydraIdentityProvider,
+            HttpsProxy httpsProxy)
+    {
+        super(standard, hydraIdentityProvider, httpsProxy);
+
+        this.binder = requireNonNull(binder, "binder is null");
+        this.hydraIdentityProvider = requireNonNull(hydraIdentityProvider, "hydraIdentityProvider is null");
+        requireNonNull(dockerFiles, "dockerFiles is null");
```
Comment (Praveen2112) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpsProxyAuthentication.java:56]: nit: rnn is redundant

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpsProxyAuthentication.java [hunk 22]
```diff
@@ -0,0 +1,91 @@
...
+        this.configDir = dockerFiles.getDockerFilesHostDirectory("conf/environment/singlenode-oauth2-https-proxy-authn/");
+        this.httpsProxy = requireNonNull(httpsProxy, "httpsProxy is null");
+    }
+
+    @Override
+    public void extendEnvironment(Environment.Builder builder)
+    {
+        builder.configureContainer(COORDINATOR, dockerContainer -> {
+            dockerContainer
+                    .withCopyFileToContainer(
```
Comment (Praveen2112) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpsProxyAuthentication.java:66]: Can we inline `forHostPath`

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java [hunk 23]
```diff
@@ -0,0 +1,86 @@
...
+                .getDockerFilesHostDirectory("conf/environment/singlenode-oauth2-http-proxy-authn/");
```
Comment (Praveen2112) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxyAuthentication.java:56]: Can we have the directory name which mimics the class name ?

**Review by Praveen2112 — COMMENTED** [review]
LGTM

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpsProxy.java [hunk 24]
```diff
@@ -40,9 +42,14 @@ public class EnvSinglenodeOauth2HttpsProxy
...
-    public EnvSinglenodeOauth2HttpsProxy(DockerFiles dockerFiles, PortBinder binder, Standard standard, HydraIdentityProvider hydraIdentityProvider)
+    public EnvSinglenodeOauth2HttpsProxy(
+            DockerFiles dockerFiles,
+            PortBinder binder,
+            Standard standard,
+            HydraIdentityProvider hydraIdentityProvider,
+            HttpProxy httpProxy)
     {
-        super(ImmutableList.of(standard, hydraIdentityProvider));
+        super(standard, hydraIdentityProvider, httpProxy);
```
Comment (krvikash) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpsProxy.java]: Why did we remove `ImmutableList.of` here?

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxy.java [hunk 25]
```diff
@@ -40,9 +41,14 @@ public class EnvSinglenodeOauth2HttpProxy
...
     @Inject
-    public EnvSinglenodeOauth2HttpProxy(DockerFiles dockerFiles, PortBinder binder, Standard standard, HydraIdentityProvider hydraIdentityProvider)
+    public EnvSinglenodeOauth2HttpProxy(
+            DockerFiles dockerFiles,
+            PortBinder binder,
+            Standard standard,
+            HydraIdentityProvider hydraIdentityProvider,
+            HttpProxy httpProxy)
     {
-        super(ImmutableList.of(standard, hydraIdentityProvider));
```
Comment (krvikash) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxy.java:45]: Why did we remove `ImmutableList.of` here?

**Review by krvikash — APPROVED** [review]

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpsProxy.java (hunk 24 above)
Comment (marcinsbd) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpsProxy.java]: Good catch. 
I guess the EnvironmentProvider already creates immutable copy of arguments so I decided there is no sense for adding one more wrapper.
@Praveen2112 @ssheikin wdyt?

Comment (ssheikin) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpsProxy.java]:
```
    protected EnvironmentProvider(EnvironmentExtender... bases)
    {
        this(ImmutableList.copyOf(requireNonNull(bases, "bases is null")));
    }
```

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxy.java (hunk 25 above)
Comment (ssheikin) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxy.java:45]: https://github.com/trinodb/trino/pull/22934#discussion_r1743324145

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpsProxy.java (hunk 24 above)
Comment (krvikash) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpsProxy.java]: Then can we have a prep commit for this? As removing `ImmutableList.of` is not related to the current commit.

Comment (marcinsbd) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpsProxy.java]: done

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxy.java [hunk 26]
```diff
@@ -13,7 +13,6 @@
  */
```
Comment (ssheikin) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxy.java:13]: long commit message

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpsProxy.java [hunk 27]
```diff
@@ -42,7 +41,7 @@ public class EnvSinglenodeOauth2HttpsProxy
     @Inject
     public EnvSinglenodeOauth2HttpsProxy(DockerFiles dockerFiles, PortBinder binder, Standard standard, HydraIdentityProvider hydraIdentityProvider)
     {
-        super(ImmutableList.of(standard, hydraIdentityProvider));
+        super(standard, hydraIdentityProvider);
```
Comment (Praveen2112) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpsProxy.java]: I would leave it as it is and would open up a new PR which would fix it for all the environment. WDYT ?

Comment (marcinsbd) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpsProxy.java]: let's do it.

**Review by krvikash — APPROVED** [review]
